magma/protocol_type.py

```diff
@@ -34,6 +34,16 @@ def flip(cls):
     def __len__(cls):
         return len(cls._to_magma_())
 
+    @property
```

Comment:
I feel like I am starting to agree with @cdonovick ... rather than just adding all these functions directly to this type (ala mixins) we should just do `value = magma_value(value)` everywhere, then we don't have to duplicate these functions on the protocol types (ala protocols).

Reply:
ok, let me try to use `magma_value` more to avoid these property/methods.

Reply:
deleted